xkcdpass/xkcd_password.py

```diff
@@ -368,11 +368,7 @@ def randomized_delimiter_join(words, delimiters=DEFAULT_DELIMITERS):
     Join the words into a password with random delimiters between each word
     """
 
-    final_passwd = ''
-    for word in words:
-        final_passwd += choose_delimiter(delimiters) + word
-
-    return final_passwd + choose_delimiter(delimiters)
+    return ''.join(map(lambda w: w + choose_delimiter(delimiters), words))[:-1]
```

Comment:
This doesn't work when the user provides a delimiter that is longer than one character.

Something like this should work:

```python
final_passwd = words[0]
for word in words[1:]:
    final_passwd += choose_delimiter(delimiters) + word
return final_passwd
```
also I think it's easier to read :)

The test should still work as the random delimiters are always one character long.

Reply:
The `delimiters` variable is set by the `-D` option by the user.

```
def choose_delimiter(delimiters):
    """
    Choose a random delimiter from the list
    """
    return rng().choice(delimiters)
```

If the user uses both the `-R` and the `-D` options, each character in the `-D` options is chosen independently. There's no way for users to specify they want to use multiple characters with random delimiters.

This is the output of the current public version.

```
> xkcdpass -n3 --min 3 --max 3 -R -D'-+*/'
+oaf-mudgab+
```

If we want to add multi-character random delimiters in the future, we can deal with it later. But I think using multiple characters for random delimiter is equivalent to using a larger `--max` option value with single character random delimiters.

Reply:
ohhh right!! I missed that bit! my bad ;)